### ISSUE TITLE
forecast evaluation time 

### ISSUE THREAD

**Issue by juniperlsimonis** [weecology/portalPredictions#157]
There are some open questions in the evaluation of time series models, especially in the context of iterated forecasting, that I think we can use our project to help inform. I'd like to start having a general discussion about this topic, so please feel free to bounce ideas around and help me brainstorm and plan out where we're going with this!
[this is definitely a brainstorm, so I apologize for the length and tangents. note the bolded questions though! I could definitely use specific answers to those]

The overarching issue is needing to have a general quantitative measure of model predictive ability. We want this measure so that we can rank and compare models and for the purposes of combining models into an ensemble (the measure of model ability forms the basis of model weights in the ensemble).

For the metrics, a lot of the forecasting literature I've seen has primarily used error-based measures like RMSE and the like. However, most all of those (or maybe all of them?) have major distributional assumptions built in that we should avoid for census of small populations. Namely, they assume symmetric and un-bounded distributions. Further, those methods generally assume a fixed point prediction, and therefore don't include model uncertainty. We can avoid those issues, however, if we instead use a predictive probability evaluation, where we quantify the probability of observing the new data based on the model fit to that point. (This will mean we'll need to store model output in a different format than it current is being stored)

In the context of any model, the most proper way to evaluate performance is on data that were held out of the data used to fit the model. That is, the model is fit on one data set (training set) and then used to predict another data set (test set), and its ability to predict the test set says something about the capacity of the model to make predictions. For time series models, the ability to predict into the future is what we're interested in. A better model then is better able to predict future states (rather than necessarily being better able to fit historic data). 

I'm currently thinking about this in the context of a "rolling forecast origin" approach (https://www.otexts.org/fpp/2/5) where the long-running ability of the model to forecast the data point X time steps ahead is what's used to evaluate the model. 

So then the natural question is **"how many time steps ahead do we want to optimize for?"** (alternatively stated, "What time scale of forecast are we predominately interested in?"). this is important because some models may be good at predicting a month out but relatively horrible at predicting next year. Other models may be less good at next month but great at next year. 

From an empirical/natural history perspective (@gmyenni @skmorgane), what is the most relevant time point for optimizing the forecast? I would think from a practical/study stand point it would be that we'd want to optimize the next census (rather than multiple censuses in advance), but maybe not?

This is functionally similar to the approach of sequential Monte Carlo (i.e. the particle filter), and that relationship has made me wonder what the role of this kind of evaluation is in the context of particle filter models. That is, the likelihood estimation is based on one-step-ahead forecasting and evaluation, so that seems to align with the rolling forecast origin approach. 

However, in SMC, and in general with validation approaches, including rolling forecast validation, there's an equal weighting of all observations, which brings me to another question: **"what is the time window for which we want to evaluate the models?"** ostensibly, aside from the initial few data points (needed for the bare minimum degrees of freedom), we could evaluate the model on all of our data points. So how far back do we go? This is similar to asking about the scope of the training vs testing data, but specific for time series ("when do we start testing the ability of the model to forecast?")

We could go all the way back to the beginning of the data collection. Or we could use just since the beginning of the forecasting project. Or really any time window.
